src/sampling.jl

```diff
@@ -527,22 +562,46 @@ function sample!(a::AbstractArray, wv::WeightVec, x::AbstractArray;
 end
 
 sample{T}(a::AbstractArray{T}, wv::WeightVec, n::Integer; replace::Bool=true, ordered::Bool=false) =
-    sample!(a, wv, Array(T, n); replace=replace, ordered=ordered)
+    sample!(a, wv, Array{T}(n); replace=replace, ordered=ordered)
 
 sample{T}(a::AbstractArray{T}, wv::WeightVec, dims::Dims; replace::Bool=true, ordered::Bool=false) =
-    sample!(a, wv, Array(T, dims); replace=replace, ordered=ordered)
+    sample!(a, wv, Array{T}(dims); replace=replace, ordered=ordered)
+
 
 # wsample interface
 
-wsample(w::RealVector) = sample(weights(w))
-wsample(a::AbstractArray, w::RealVector) = sample(a, weights(w))
+"""
+    wsample!(a, w, x, [replace=true,] [ordered=false])
 
+Select a weighted sample from an array `a` and store the result in `x`. The
+weights are given in `w`. This function accepts two keyword arguments:
+
+  • `replace`: Should sampling be performed with replacement? Defaults
+    to `true`.
+  • `ordered`: Should an ordered sample be taken? Defaults to `false`.
+"""
 wsample!(a::AbstractArray, w::RealVector, x::AbstractArray; replace::Bool=true, ordered::Bool=false) =
     sample!(a, weights(w), x; replace=replace, ordered=ordered)
 
+
+"""
+    wsample([a,] w, [n,] [dims,] [replace=true,] [ordered=false]) -> Array
+
+Select a random sample of the weights given in `w` if no array `a` is given.
```

Comment:
I don't understand the meaning of the one-argument form. Maybe don't document it as it's confusing, and we'll deprecate it soon?

Same remark as above about `n`/`dims`.